lib/ipfs-pinner.js

```diff
@@ -0,0 +1,46 @@
+const fs = require('fs')
+const path = require('path')
+const { client, PinnerError } = require('@aragonone/ipfs-pinner-client')
+
+const ALREADY_UPLOADED_ERROR = 'File is already uploaded with cid '
+const IPFS_PINNER_ENDPOINT = 'https://ipfs-pinner.backend.aragon.org'
+
+function requiresIpfsUpload(justification) {
+  return !justification.startsWith('ipfs:') && justification.endsWith('.md')
+}
+
+async function ipfsUpload(disputePath, justification, sender) {
+  let content
+
+  if (justification.endsWith('.md')) {
+    const justificationPath = path.join(process.cwd(), disputePath, justification)
+    if (!fs.existsSync(justificationPath)) throw Error(`Justification path "${justificationPath}" does not exit`)
+    console.log(`Uploading justification "${justification}" to IPFS...`)
+    content = justificationPath
+  } else {
+    console.log(`Uploading justification "${justification}" as plain text (only markdown is supported for IPFS evidence)...`)
+    content = justification
+  }
+
+  await client.setEndpoint(IPFS_PINNER_ENDPOINT)
+
+  let cid
+  try {
+    const response = await client.upload(sender, content)
+    cid = response.cid
+    console.log(`Uploaded file ${cid}`)
+  } catch (error) {
+    if (!(error instanceof PinnerError)) throw error
+    const message = error.errors[0].file
+    if (!message || !message.includes(ALREADY_UPLOADED_ERROR)) throw error
+    cid = message.replace(ALREADY_UPLOADED_ERROR, '')
```

Comment:
This seems kind of sketchy from the API side, but 🤷 .

The alternative would be to attach the CID to the error directly.

Reply:
yes it is, but I wasn't expecting this snippet to be sth elegant haha

Reply:
Yeah, I was thinking that on this error you'd almost always want to know the already-updated CID